### ISSUE TITLE
Automatic enemy targeting by party members.

### ISSUE THREAD

**Issue by EvgenijM86** [elonafoobar/elonafoobar#1478]
I noticed that ally do not attack enemies, unless:
1) you attack the enemy
2) enemy attacks player character
3) you target the enemy with l or *

a simple way to enable auto targeting is to add find_enemy_target() after this function:
```
optional<TurnResult> handle_pc_action(std::string& action)
{
    find_enemy_target();
```
That way at the beginning of your turn you have already acquired your target. And the next turn is for your party member, who will copy your enemy_id and use it to do attacks. This is purely quality of life change, because in principle you can manually target enemy at the beginning of every turn and get the same effect.

I can create a pull request that add a config option that enables auto-target mode for party members, if you want.

**Comment by EvgenijM86** [comment]
And party members will still not respond if they are attacked by the enemy and you did not target that enemy (i.e. enemy is outside of your visibility). I am not sure if this should be fixed or not, because it will affect balance, potentially making your party more powerful for grinding dungeons, as they could grind on their own (if powerful enough).

**Comment by ki-foobar** [comment]
The issue is one of the most significant changes between vanilla and variants which have improved pets combat QoL: allies don't target enemies automatically.

> This is purely quality of life change, because in principle you can manually target enemy at the beginning of every turn and get the same effect.

I thought that fixing the issue affected the game balance before, but now I agree with you, it is purely QoL change. You can manually targe t your enemy every turn.


> And party members will still not respond if they are attacked by the enemy and you did not target that enemy (i.e. enemy is outside of your visibility). I am not sure if this should be fixed or not, because it will affect balance, potentially making your party more powerful for grinding dungeons, as they could grind on their own (if powerful enough).

Some variants have improved the pet AI (variant omake and many others) and allies in such variants are smarter and more powerful than vanilla. Kinds of things which affects game balance should be achieved as mods (Currently some game changing features are implemented as built-in though). Elona foobar (will) provides the means to improve pet AI by modding, but won't change them at all without mods.

**Comment by EvgenijM86** [comment]
So, should I create a pull request of auto targeting of enemies that are visible by player character? I can add it as a config option like with digital clock.

**Comment by ki-foobar** [comment]
That would be great, thank you.

**Comment by ki-foobar** [comment]
implemented by #1482
